tests/backends.py

```diff
@@ -0,0 +1,19 @@
+from collections import OrderedDict
+from django.core.cache.backends.locmem import LocMemCache
+
+try:
+    # For Django versions < 2.1
+    from django.utils.synch import RWLock as Lock
+except ImportError:
+    from threading import Lock
+
+
+class ThreadLocalCache(LocMemCache):
+    def __init__(self, *args, **kwargs):
+        self._cache = OrderedDict()
+        self._expire_info = {}
+        self._lock = Lock()
+
+        # Skip calling the parent here, we've initialised everything
+        # that's needed already.
+        super(LocMemCache, self).__init__({})
```

Comment:
I'm baffled that this could ever work and not cause an error in testing. 
I looked at my Django 2.0.9 code (first available) and it looked like this:

```python
class LocMemCache(BaseCache):
    def __init__(self, name, params):
        BaseCache.__init__(self, params)
        self._cache = _caches.setdefault(name, {})
        self._expire_info = _expire_info.setdefault(name, {})
        self._lock = _locks.setdefault(name, RWLock())
``` 

So the `__init__` method expects **2** arguments. But you're only passing it **1**. How can that work?

Reply:
Also, you're setting (for example) `self._expire_info = ...` first and then calling the `super().__init__` which will *override* `self._expire_info`. So all of these `self._cache = ...`, `self._expire_info = ...`, and `self._lock = ...` are surely all "in vain". 
Yet it didn't seem to break the tests.

Reply:
Maybe my code comment isn't very helpful.

The call `super(LocMemCache, self).__init__({})` is equivalent to `BaseCache.__init__(self, {})`, not `LocMemCache.__init__(self, {})`.

So essentially we are skipping the `__init__` of `LocMemCache` and going straight to each ancestor (but keeping the rest of the class).

Reply:
I've updated the test backend with a docstring and comment.

Reply:
I don't quite agree, it seems like an implicit, unnecessary step to first pollute the global cache used by LocMem with an entry that's not really going to be used. But this is just a test backend anyway, so whatever way you prefer.